test/test_oneview_server_hardware_type.py

```diff
@@ -0,0 +1,200 @@
+###
+# Copyright (2016) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+import unittest
+import mock
+import yaml
+
+from hpOneView.oneview_client import OneViewClient
+from oneview_server_hardware_type import ServerHardwareTypeModule, SERVER_HARDWARE_TYPE_DELETED, \
+    SERVER_HARDWARE_TYPE_ALREADY_ABSENT, SERVER_HARDWARE_TYPE_ALREADY_UPDATED, SERVER_HARDWARE_TYPE_UPDATED, \
+    SERVER_HARDWARE_TYPE_NOT_FOUND
+
+FAKE_MSG_ERROR = 'Fake message error'
+
+RACK_URI = '/rest/racks/rackid'
+
+YAML_SERVER_HARDWARE_TYPE = """
+        config: "{{ config }}"
+        state: present
+        data:
+          name: 'My Server Hardware Type'
+          description: "New Description"
+          """
+
+YAML_SERVER_HARDWARE_TYPE_CHANGE = """
+        config: "{{ config }}"
+        state: present
+        data:
+          name: 'My Server Hardware Type'
+          newName: 'My Server Hardware Type'
```

Comment:
Why are you providing the same value for name and newName? I suggest you change to a different name, or remove the newName from the yaml.